pkg/config/validation/validation.go

```diff
@@ -1514,21 +1523,55 @@ func ValidateAuthorizationPolicy(_, _ string, msg proto.Message) error {
 }
 
 // ValidateRequestAuthentication checks that request authentication spec is well-formed.
-func ValidateRequestAuthentication(_, _ string, msg proto.Message) error {
+func ValidateRequestAuthentication(name, namespace string, msg proto.Message) error {
 	in, ok := msg.(*security_beta.RequestAuthentication)
 	if !ok {
 		return errors.New("cannot cast to RequestAuthentication")
 	}
-	// TODO(diemtvu) add more details validation.
+
 	var errs error
+	errs = appendErrors(errs, validateWorkloadSelector(in.Selector))
+
 	for _, rule := range in.JwtRules {
-		if len(rule.Issuer) == 0 {
-			errs = appendErrors(errs, fmt.Errorf("issuer must be set"))
-		}
+		errs = appendErrors(errs, validateJwtRule(rule))
 	}
 	return errs
 }
 
+func validateJwtRule(rule *security_beta.JWT) (errs error) {
+	if rule == nil {
+		return nil
+	}
+	if rule.Issuer == "" {
+		errs = multierror.Append(errs, errors.New("issuer must be set"))
+	}
+	for _, audience := range rule.Audiences {
+		if audience == "" {
+			errs = multierror.Append(errs, errors.New("audience must be non-empty string"))
+		}
+	}
+
+	if len(rule.JwksUri) != 0 {
+		// TODO: do more extensive check (e.g try to fetch JwksUri)
+		if _, err := security.ParseJwksURI(rule.JwksUri); err != nil {
+			errs = multierror.Append(errs, err)
+		}
+	}
+
+	for _, location := range rule.FromHeaders {
+		if len(location.Name) == 0 {
```

Comment:
why use len here and == "" in other places?

Reply:
Cleaned up.